src/cli.ts

```diff
@@ -4,6 +4,51 @@ import * as program from 'commander';
 import { Log } from './models/Log';
 import { SimulatorConfig, run } from './index';
 
+function validateAppTypeJSONInput(input: any) {
+  if (typeof input.state !== 'object') {
+    new Log(false).error(
+      'appType custom shadow is missing object value for "state" key',
+    );
+    return false;
+  }
+
+  if (typeof input.state.reported !== 'object') {
+    new Log(false).error(
+      'appType custom shadow "state" object is missing object value for "reported" key',
+    );
+    return false;
+  }
+
+  return true;
+}
+
+const handleAppType = (input: any, _: unknown) => {
+  if (input === 'mss' || input === 'atv2') {
+    return input;
+  }
+
+  if (input[0] !== '[' && input[0] !== '{') {
+    new Log(false).error(
+      'Input for appType may only be "mss", "atv2", or a JSON-encoded object',
+    );
+    process.exit();
+  }
+
+  try {
+    input = JSON.parse(input);
+    if (!validateAppTypeJSONInput(input)) {
+      new Log(false).info(
+        `Expected input: '{"state":{"reported":{<DEVICE_DATA>}}}'`,
```

Comment:
Implies you can only have `"reported"`. You can have `"desired"` too, but it's not required.

Reply:
Would a good solution for this to be an error something like:

`Expected input: '{"state":{"reported":{<DEVICE_DATA>}, "desired?":{}}}` 

Or `Required input:  '{"state":{"reported":{<DEVICE_DATA>}}}', Optional state key of desired can be passed as well.

Reply:
Or:

```
Expected input: '{"state":{"reported":{...}, "desired":{...}}}', "desired" is optional.
```